pycalphad/core/calculate.py

```diff
@@ -301,6 +301,9 @@ def calculate(dbf, comps, phases, mode=None, output='GM', fake_points=False, bro
     if points_dict is None and broadcast is False:
         raise ValueError('The \'points\' keyword argument must be specified if broadcast=False is also given.')
     nonvacant_components = [x for x in sorted(comps) if x.number_of_atoms > 0]
+    desired_active_pure_elements = [list(x.constituents.keys()) for x in comps]
+    desired_active_pure_elements = [el.upper() for constituents in desired_active_pure_elements for el in constituents]
+    nonvacant_elements = sorted([x for x in set(desired_active_pure_elements) if x != 'VA'])
```

Comment:
Needs an import `from pycalphad.core.utils import get_pure_elements`

```suggestion
    nonvacant_elements = get_pure_elements(dbf, comps)
```